src/Test/Html/Selector.elm

```diff
@@ -265,6 +266,38 @@ text =
     Internal.Text
 
 
+{-| Matches elements that contain children that match the given selectors.
+You will get the element and **not** the child.
+This is useful when you want a sepecific element and continue working with it.
+You can use `Test.Html.Query.contains` if you just want to expect that an element contains some child.
+
+    import Html
+    import Html.Events exposing (onClick)
+    import Test exposing (test)
+    import Test.Html.Event as Event
+    import Test.Html.Query as Query
+    import Test.Html.Selector exposing (hasChildWith, tag)
+
+    test : Test
+    test =
+        test "..." <|
+            Html.div []
+                [ Html.button [ onClick NopeMsg ] [ Html.text "not me" ]
+                , Html.button [ onClick ClickedMsg ] [ Html.text "click me" ]
+                ]
+                |> Query.find
+                    [ tag "button"
+                    , hasChildWith [ text "click me" ]
+                    ]
```

Comment:
This example is confusing to me because I would think of the button as "having" text "click me", not as containing a child with text "click me".

Reply:
What if we limited the API and provided `withText : String -> Query` instead?

Reply:
I like that idea on first impression.  Though I'm not sure I completely understand the intended distinction between queries and selectors.

Reply:
I definitely had the use-case where I needed `hasChildWith` without just checking text in the children. I see that `withText`might be nice as well though. I will think of a better example.